### PR TITLE
refactor: move loader to state and rename game

### PR DESCRIPTION
**Pull request by yaahallo**
Prior to this change, I had mixed up the idea of states and systems and
had mistakenly classified Loader as a system. It is a state, and using
it just for the loading step is probably not a good design.

This change moves the loader to a new directory called state, which is
where game states will go. It renames it to game.rs, and it will now
represent the main game state. It still does loading in its on_start,
this might end up getting moved elsewhere, like setup for the various
systems, which might help with the issue currently with systems running
before all the assets are loaded.

### CODE REVIEW

File: game-core/src/state/game.rs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    use amethyst::{
+        core::{Parent, Transform},
+        ecs::Entity,
+        prelude::*,
+        renderer::{Camera, SpriteRender, SpriteSheetHandle, Transparent},
+    };
+    // Initialize a background
+    pub fn background_sprite(world: &mut World, sprite_sheet: &SpriteSheetHandle) -> Entity {
+        let mut transform = Transform::default();
+        transform.translation.z = -10.0;
```
Comment (GabeVillalobos) [inline game-core/src/state/game.rs]: Magic Numbers. Pull this into a const or config maybe?

File: game-core/src/state/game.rs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            .create_entity()
+            .with(transform)
+            .with(sprite)
+            .with(Transparent)
+            .build()
+    }
+
+    pub fn camera(world: &mut World, parent: Entity) {
+        let mut transform = Transform::default();
+        transform.translation.z = 2.0;
```
Comment (GabeVillalobos) [inline game-core/src/state/game.rs]: same

File: game-core/src/state/game.rs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            flip_horizontal: false,
+            flip_vertical: false,
+        };
+        world.create_entity().with(transform).with(sprite).build()
+    }
+
+    // Initialize a sprite as a reference point at a fixed location
+    pub fn reference_sprite(world: &mut World, sprite_sheet: &SpriteSheetHandle) -> Entity {
+        let mut transform = Transform::default();
+        transform.translation.x = 100.0;
```
Comment (GabeVillalobos) [inline game-core/src/state/game.rs]: magic number

File: game-core/src/state/game.rs (hunk 1 above)
Comment (yaahallo) [reply game-core/src/state/game.rs]: the background sprite isnt actually in use I think. That was part of the example, I should honestly remove it. I'll do that as a separate commit

File: game-core/src/state/game.rs (hunk 2 above)
Comment (yaahallo) [reply game-core/src/state/game.rs]: This one can be redefined as a const.

File: game-core/src/state/game.rs (hunk 3 above)
Comment (yaahallo) [reply game-core/src/state/game.rs]: this one is arbitrary and due for removal.

**Pull request merged**